src/components/sbb-button/sbb-button.tsx

```diff
@@ -100,29 +113,29 @@ export class SbbButton {
           this.click.emit(this.eventId);
         }
       },
-      type: this.type,
+      type: this.buttonType,
       value: this.value,
     };
 
     const finalAttributes = this.visualButtonOnly ? mainAttributes : buttonAttributes;
 
     return (
       <TAGNAME {...finalAttributes}>
-        {this.icon && hasNoLabel && this.iconDescription ? (
+        {this.icon && hasNoLabel && this.iconDescription && (
           <span class="button__icon-description">{this.iconDescription}</span>
-        ) : (
-          ''
         )}
 
-        {this.icon === true ? (
+        {this.icon && (
           <span class="button__icon">
-            <slot />
+            <slot name="icon" />
```

Comment:
I would also suggest adding an additional check to prevent the empty slot from being rendered if an icon is not provided:
```suggestion
            <slot name="icon">
              {this.iconName && <sbb-icon name={this.iconName}><sbb-icon/>}
            </slot>
```